src/Compiler/Checking/TypeRelations.fs

```diff
@@ -117,16 +117,16 @@ let rec TypeFeasiblySubsumesType ndeep g amap m ty1 canCoerce ty2 =
 
     | _ -> 
         // F# reference types are subtypes of type 'obj' 
-        (isObjTy g ty1 && (canCoerce = CanCoerce || isRefTy g ty2)) 
-        ||
-        (isAppTy g ty2 &&
-         (canCoerce = CanCoerce || isRefTy g ty2) && 
-         begin match GetSuperTypeOfType g amap m ty2 with 
-         | None -> false
-         | Some ty -> TypeFeasiblySubsumesType (ndeep+1) g amap m ty1 NoCoerce ty
-         end ||
-         ty2 |> GetImmediateInterfacesOfType SkipUnrefInterfaces.Yes g amap m 
-             |> List.exists (TypeFeasiblySubsumesType (ndeep+1) g amap m ty1 NoCoerce))
```

Comment:
Before all these were executed "lazily", if left side of OR was true. Now they all get executed unconditionally, which is more expensive.

Reply:
Yeah I’m  just trying to debug this. Will put improve/revert it once I know what is going on